MobileCenter/MobileCenter/Internals/AppDelegate/MSAppDelegateForwarder.m

```diff
@@ -179,17 +198,40 @@ + (IMP)swizzleOriginalSelector:(SEL)originalSelector
      * depend on the SDK return value for its own logic so customers already have to call the SDK API
      * in their implementation which makes swizzling useless.
      */
-    if (![self.selectorsNotToOverride containsObject:originalSelectorString]) {
+    if (![self.selectorsNotToOverride containsObject:originalSelectorStr]) {
       originalImp = method_setImplementation(originalMethod, customImp);
+    } else {
+      warningMsg =
+          [NSString stringWithFormat:@"This selector is not supported when already implemented. %@", remediationMsg];
     }
   } else if (![originalClass instancesRespondToSelector:originalSelector]) {
 
-    /*
-     * The original class may not implement the selector (e.g.: optional method from protocol),
-     * add the method to the original class and associate it with the custom implementation.
-     */
-    Method customMethod = class_getInstanceMethod(self, customSelector);
-    methodAdded = class_addMethod(originalClass, originalSelector, customImp, method_getTypeEncoding(customMethod));
+    // Check for deprecation.
+    NSString *deprecatedSelectorStr = self.deprecatedSelectors[originalSelectorStr];
+    if (deprecatedSelectorStr &&
+        [originalClass instancesRespondToSelector:NSSelectorFromString(deprecatedSelectorStr)]) {
+
+      // An implementation for the deprecated selector exists. Don't add the new method, it might eclipse the original
+      // implementation.
+      warningMsg = [NSString
+          stringWithFormat:
+              @"No implementation found for this selector, though an implementation of its deprecated API '%@' exists.",
```

Comment:
Why are you saying `No implementation found for this selector` here? The above comment says it found an implementation of the deprecated selector. Am I missing something?

Reply:
An implementation for the deprecated selector has been found but no implementation were found for the current API which is the new API supposed to replace the deprecated one ;).

Reply:
I'll rephrase the comment.

Reply:
Maybe we can be more explicit here and add a suggestion, e.g. "Please implement it's successor" or something like that.